cylc/flow/job_pool.py

```diff
@@ -152,37 +169,36 @@ def set_job_time(self, job_d, event_key, time_str=None):
 
         Set values of both event_key + "_time" and event_key + "_time_string".
         """
-        point, name, sub_num, _ = self.parse_job_item(job_d)
+        update_time = time()
+        point, name, sub_num = self.parse_job_item(job_d)
         j_id = (
             f'{self.workflow_id}{ID_DELIM}{point}'
             f'{ID_DELIM}{name}{ID_DELIM}{sub_num}')
         try:
             setattr(self.pool[j_id], event_key + '_time', time_str)
+            self.pool[j_id].stamp = f"{j_id}@{update_time}"
```

Comment:
In case you have not noticed... Most job messages already have an event time. Check the method signature of `process_message` in `cylc.task_events_mgr`. There should be an `event_time` argument. You may want to pass that into the logic here and above.

Reply:
The problem is I need sub-second time (which `event_time` isn't), and at present it's seconds as float, i.e.:
```
        {
            "stamp": "sutherlander|noz|20190101T00|baa|1@1569226838.5164254",
            "id": "sutherlander|noz|20190101T00|baa|1",
            "submitNum": 1,
            "state": "running",
.
.
.
```

Reply:
The `stamp` is for internal use really, it's so we can tell that a data element has been updated (usually between loops).

Reply:
In which case, maybe worth adding some comments to explain this?

Reply:
Agreed - `stamp` will be the topic of the PR associated with [incremental UIS data sync](https://github.com/cylc/cylc-flow/issues/3207)